tooling/azure-automation/resources-cleanup/src/resources_cleanup.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+import automationassets
+from typing import List
+import datetime
+
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.resource.resources.v2022_09_01.models._models_py3 import GenericResourceExpanded, ResourceGroup
+
+
+# If DRY_RUN is TRUE, the script will print which resource groups should be deleted
+# without deleting them. If it is FALSE, the script will print which resource groups 
+# should be deleted and delete those that should be deleted.
+DRY_RUN = True
+
+# VERBOSE is used to control whether to print all the resources of each resource group
+# for informational purposes.
+VERBOSE = False
+
+DEFAULT_API_VERSION = "2022-09-01"
+
+
+def get_date_time_from_str(date_time_str: str)-> datetime.datetime:
+    """ get_date_time_from_str expects an input date in ISO 8601 with Z suffix
+    e,g: 2024-01-26T17:08:13.8139962Z
+    In Python < 3.11, fromisoformat() does not accept Z suffix even if it is valid in
+    ISO 8601 (https://discuss.python.org/t/parse-z-timezone-suffix-in-datetime/2220).
+    """
+    date_time_str = date_time_str.replace("Z", "").replace("z", "")
+    dot = "."
+    if dot in date_time_str:
+        dot_index = date_time_str.index(dot)
+        date_time_str = date_time_str[0:dot_index]
+    
+    utc_suffix = '+00:00'
+    date_time_str += utc_suffix
+    
+    return datetime.datetime.fromisoformat(date_time_str)
+
+
+def time_delta_greater_than_two_days(now: datetime.datetime, resource_group_creation_time: datetime.datetime):
+    if now == None:
+        print("now time is None")
+        return False
+    
+    if resource_group_creation_time == None:
+        # We do not want to delete the resource group if it does not have a creation timestamp.
+        print("resource_group_creation_time is None")
```

Comment:
should we print the comment out instead of the conditional statement evaluation? The comment bring printed would be a better indicator to the person reading through the logs. WDYT?

Reply:
Yes, you are right. Doing it now. Thanks